spec/javascripts/unit/admin/reports/controllers/bulk_coop_controller_spec.js.coffee

```diff
@@ -0,0 +1,34 @@
+describe "bulkCoopCtrl", ->
+  ctrl = $httpBackend = null
+  scope = controller = {}
+  reportType = 'supplier_report'
+  Report = {
+    gridOptions: ->
+      onRegisterApi: ->
+    columnOptions: ->
+  }
+
+  beforeEach ->
+    scope = {}
+
+    module('admin.reports')
+    inject ($controller, _$httpBackend_) ->
+      $httpBackend = _$httpBackend_
+      ctrl = $controller 'bulkCoopCtrl', {$scope: scope, BulkCoopReport: Report, distributors: [], reportType: reportType }
+
+  it "init ui-grid loading flags should be false", ->
+    expect(scope.loading).toBe false
+    expect(scope.loadAttempted).toBe false
+
+  it "should set default reportType", ->
+    expect(scope.reportType).toEqual 'supplier_report'
+
+  it "can load reports", ->
+    $httpBackend.expectGET("/admin/reports/bulk_coop.json").respond 200
```

Comment:
Would be good to return mock response here and test basic assignment of data to the appropriate scope variables.